### ISSUE TITLE
Only 99.97% verified after creating then verifying large (~500 GB) folder

### ISSUE THREAD

**Issue by alimony** [transmission/transmission#5112]
### What is the issue?

I'm trying to create a torrent file of a full disk backup that I have in a folder, then seed it so I can sync it to another remote computer reliably. However, I keep running into an issue where Transmission will not verify the local content fully, after creating the torrent file, and then immediately verify it.

The folder in question is approximately 500 GB of varied content (including other folders etc.), and the torrent file creation runs without issues. I then add the torrent file so I can start seeding, and the local content is verified, but only up until 99.97% or so. Looking at what has been verified, random little files are just not found, and I see no pattern in the kind of file.

I've tried this with several different folders ranging from ~300 GB to ~700 GB, and the same thing always happens. I also tried creating the torrent file with mktorrent instead, and the result was the same. It seems like something goes wrong with how Transmission verifies local data. I've tried with both Transmission 3.0.0 stable and 4.0.1 stable.

I really wish I could get this working, as there aren't really many good ways to sync hundreds of gigabytes over the Internet with intermittent access and varying speeds. The beauty of using torrent for this is that I know I will eventually end up with the correct data, exactly because of the checksumming.

### Which application of Transmission?

macOS app

### Which version of Transmission?

4.0.1

**Comment by xavery** [comment]
Sounds similar to #2377 .

**Comment by Coeur** [comment]
>Looking at what has been verified, random little files are just not found, and I see no pattern in the kind of file.

Could you figure out if those missing files:
- were on disk previously, but aren't anymore on the disk?
- if they are still on disk, did their last modification date changed recently?
- are symlinks or not?
- are hidden files or not? (like ".DS_Store")
- are zero bytes files?

Also:
- which filesystem is this disk? (Applications > Utilities > Disk Utility > check if it's APFS, exFat, NTFS, FAT, ...)

**Comment by ckerr** [comment]
@alimony I'm not sure what could be causing this. How does 4.0.2 look?

**Comment by GaryElshaw** [comment]
I'm wondering if this might be related to the sym-link issue last week?

**Comment by tearfur** [comment]
Trailing or leading spaces in file/directory names could be a reason. `transmission-create` trims these path names.
Can confirm that Transmission cannot find those files after adding the torrent.

```console
$ ll -R test-torrent/
test-torrent/:
total 4
-rw-r--r--. 1 hyy hyy  5 Oct 16 16:59  a.txt
drwxr-xr-x. 1 hyy hyy 10 Oct 16 16:52 ' with-space'

'test-torrent/ with-space':
total 0
-rw-r--r--. 1 hyy hyy 0 Oct 16 16:52 b.txt
$ transmission-create test-torrent/
WARNING: no trackers specified
Creating torrent "/home/test/test-torrent.torrent"
WARNING
filename "test-torrent/ with-space/b.txt" may not be portable on all systems.
consider "test-torrent/with-space/b.txt" instead.
2 files, 0.01 kB
1 piece, 32.77 kB
 done!
$ transmission-show test-torrent.torrent
Name: test-torrent
File: test-torrent.torrent

GENERAL

  Name: test-torrent
  Hash v1: 7a953fe5c8fcc0894f132cb09a0ed552b14478e1
  Created by: Transmission/4.0.4 (24077e3511)
  Created on: Mon Oct 16 17:05:16 2023

  Piece Count: 1
  Piece Size: 32.00 KiB
  Total Size: 0.01 kB
  Privacy: Public torrent

TRACKERS

FILES

  test-torrent/a.txt (0.01 kB)
  test-torrent/with-space/b.txt (0.00 kB)

```

**Comment by Coeur** [comment]
Since the original poster didn't reply in 8 months, I'll take tearfur recent finding as defining the scope of this issue.
So once we fix the whitespace trimming in filenames, let's close this issue.

**Comment by tearfur** [comment]
More precisely, I think `transmission-create` should not sanitize the filename in any way.

**Comment by Coeur** [comment]
From my test on main branch on macOS, @tearfur : `transmission-create` keeps the spaces. It's `transmission-show` (and Transmission.app) which are the culprits:
parse_torrent_file -> parse_benc -> benc::parse -> handler.String -> makeSubpathPortable

**Comment by Coeur** [comment]
We've addressed filenames with dots, spaces and other special characters. Now I consider that this issue will be fully addressed after #6194 (for empty files) gets merged.
